package rpcprotocol;

enum rpc_message_type {
  REQUEST = 0;
  RESPONSE = 1;
};

message RpcMessage {
  required rpc_message_type rpc_type = 1;
  required int32 message_id = 2;
  required string args = 3;
  optional string service = 4;
  optional string method = 5;
};

